src/App.jsx

```diff
@@ -1,48 +1,239 @@
-// App.js
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
 import TodoForm from './features/TodoForm';
 import TodoList from './features/TodoList'
 
 function App() {
   const [todoList, setTodoList] = useState([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState("");
+  const [isSaving, setIsSaving] = useState(false); 
+  const [isUpdating, setIsUpdating] = useState(false); 
 
-  const addTodo = (newTodoTitle) => {
-    setTodoList((prevTodoList) => [
-      ...prevTodoList,
-      { id: Date.now(), title: newTodoTitle, isCompleted: false }
-    ]);
+  const url = `https://api.airtable.com/v0/${import.meta.env.VITE_BASE_ID}/${import.meta.env.VITE_TABLE_NAME}`;
+
+  const token = `Bearer ${import.meta.env.VITE_PAT}`;
+
+  useEffect(() => {
+    const fetchTodos = async () => {
+      setIsLoading(true);
+
+      const options = {
+        method: "GET",
+        headers: {
+          "Authorization": token,
+        },
+      };
+
+      try {
+        const resp = await fetch(url, options);
+
+        if (!resp.ok) {
+          const errorData = await resp.json();
+          throw new Error(`Error: ${resp.status} ${errorData.error.message}`);
+        }
+
+        const data = await resp.json();
+        console.log("data ===> ", data);
+
+        const transformedData = data.records.map((record) => {
```

Comment:
This could be refactored:
```js
const transformedData = data.records.map(({ id, fields: { title = '', isCompleted = false } }) => ({
  id,
  title,
  isCompleted
}));

```